hsmd/Makefile

```diff
@@ -31,11 +33,13 @@ HSMD_COMMON_OBJS :=				\
 	common/status_wiregen.o			\
 	common/hash_u5.o			\
 	common/hsm_encryption.o			\
+	common/htlc_wire.o			\
 	common/key_derive.o			\
 	common/lease_rates.o			\
 	common/memleak.o			\
 	common/msg_queue.o			\
 	common/node_id.o			\
+	common/onionreply.o			\
```

Comment:
stray?

Reply:
Unfortunately not stray:
```
/usr/bin/ld: common/htlc_wire.o: in function `failed_htlc_dup':
/home/ksedgwic/lightning-signer/c-lightning/common/htlc_wire.c:23: undefined reference to `dup_onionreply'
/usr/bin/ld: common/htlc_wire.o: in function `towire_failed_htlc':
/home/ksedgwic/lightning-signer/c-lightning/common/htlc_wire.c:150: undefined reference to `towire_onionreply'
/usr/bin/ld: common/htlc_wire.o: in function `fromwire_failed_htlc':
/home/ksedgwic/lightning-signer/c-lightning/common/htlc_wire.c:265: undefined reference to `fromwire_onionreply'
```

If we broke `simple_htlc` out onto it's own file we could fix it, I think.  Worth doing?

Reply:
ah, I see.  probably not worth it. C-L people can raise this if they think it's useful.